### PR TITLE
va-on-this-page: Rewrite to use ul instead of dl for list of links

### PR DESCRIPTION
**Pull request by Andrew565**
## Chromatic

https://1998-on-this-page-rewrite--65a6e2ed2314f7b8f98609d8.chromatic.com

---

## Description
Closes [#1998](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1998)

## QA Checklist
- [X] Component maintains 1:1 parity with design mocks
- [X] Text is consistent with what's been provided in the mocks
- [X] Component behaves as expected across breakpoints
- [x] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [X] Tab order and focus state work as expected
- [X] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [X] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [X] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots
Before:
Screenshot 2024-09-27 at 12 41 48 PM
After:
Screenshot 2024-10-02 at 11 10 15 AM

## Acceptance criteria
- [ ] QA checklist has been completed
- [X] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [X] Documentation has been updated, if applicable
- [X] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-on-this-page/va-on-this-page.tsx [hunk 1]
```diff
@@ -90,17 +90,17 @@ export class VaOnThisPage {
...
+          {h2s.map(heading => (
+            <li>
               <a href={`#${heading.id}`} onClick={handleOnClick}>
                 <va-icon icon="arrow_downward"></va-icon>
                 {heading.innerText}
               </a>
-            ))}
-          </dd>
-        </dl>
+            </li>
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-on-this-page/va-on-this-page.tsx:101]: Are there tests that this will affect in vets-website?

Comment (Andrew565) [reply packages/web-components/src/components/va-on-this-page/va-on-this-page.tsx:101]: There is one test that looks for this component: vets-website/src/applications/sco/tests/accessibility.cypress.spec.js, but the test is structured in such a way that it should still work with this re-write.

It's worth noting that there are currently 5 pages that have "fake component" versions of the va-on-this-page that all still use the dl/dd combo to display links and should probably be converted:
1. vets-website/src/applications/combined-debt-portal/debt-letters/components/DebtDetailsOnThisPageLinks.jsx
2. vets-website/src/applications/combined-debt-portal/debt-letters/components/OnThisPageLinks.jsx
3. vets-website/src/applications/combined-debt-portal/medical-copays/components/OnThisPageDetails.jsx
4. vets-website/src/applications/combined-debt-portal/medical-copays/components/OnThisPageOverview.jsx
5. vets-website/src/applications/combined-debt-portal/medical-copays/components/OnThisPageStatements.jsx

I'll write these up in a separate ticket for after this component ships.

**Review by rsmithadhoc — APPROVED** [review]

**Comment by danbrady** [comment]
@Andrew565 If the link wraps, can we have the second line align with the first like it is in "before"?

image

**Comment by Andrew565** [comment]
@danbrady Ready for re-review, updated the "after" screenshot at the top.

**Review by danbrady — APPROVED** [review]
LGTM! (And thanks for fixing the vertical alignment of the icon too!)

File: packages/web-components/src/components/va-on-this-page/va-on-this-page.css [hunk 2]
```diff
@@ -7,43 +7,49 @@
...
   list-style-type: none;
   padding-left: 0;
 }
 
-dd a {
-  display: flex;
-  align-items: baseline;
+li {
   padding: 0.5em;
+  max-width: 372px;
```
Comment (babsdenney) [inline packages/web-components/src/components/va-on-this-page/va-on-this-page.css:34]: @Andrew565 Would you be able to revert the width back to the original? My apologies for adding that feedback to this ticket. I talked to Carol about the issue she was able to move the ticket ([#951](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/951)) to address the width to Sprint 13 so we can address the width then.

Comment (Andrew565) [reply packages/web-components/src/components/va-on-this-page/va-on-this-page.css]: Done

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**